libraries/stdlib/jdk7/test/PathExtensionsTest.kt

```diff
@@ -0,0 +1,358 @@
+
+            srcFile.copyTo(dstFile, overwrite = true)
+            assertEquals(srcFile.readText(), dstFile.readText(), "copy with overwrite over empty dir")
+
+            assertTrue(Files.deleteIfExists(srcFile))
+            assertTrue(Files.deleteIfExists(dstFile))
+
+            assertFailsWith(NoSuchFileException::class) {
+                srcFile.copyTo(dstFile)
+            }
+
+            Files.createDirectory(srcFile)
+            srcFile.resolve("somefile").writeText("some content")
+            dstFile.writeText("")
+            assertFailsWith(FileAlreadyExistsException::class, "copy dir do not overwrite file") {
+                srcFile.copyTo(dstFile)
+            }
+            srcFile.copyTo(dstFile, overwrite = true)
+            assertTrue(dstFile.isDirectory())
+            assertTrue(dstFile.listDirectoryEntries().isEmpty(), "only directory is copied, but not its content")
+
+            assertFailsWith(FileAlreadyExistsException::class, "copy dir do not overwrite dir") {
+                srcFile.copyTo(dstFile)
+            }
+
+            srcFile.copyTo(dstFile, overwrite = true)
+            assertTrue(dstFile.isDirectory())
+            assertTrue(dstFile.listDirectoryEntries().isEmpty(), "only directory is copied, but not its content")
+
+            dstFile.resolve("somefile2").writeText("some content2")
+            assertFailsWith(FileAlreadyExistsException::class, "copy dir do not overwrite non-empty dir") {
+                srcFile.copyTo(dstFile, overwrite = true)
+            }
+        } finally {
+            srcFile.toFile().deleteRecursively()
+            dstFile.toFile().deleteRecursively()
+        }
+    }
+
+    @Test
+    fun copyToNameWithoutParent() {
+        val currentDir = Paths.get("").toAbsolutePath()
+        val srcFile = Files.createTempFile(null, null)
+        val dstFile = Files.createTempFile(currentDir, null, null)
+        try {
+            srcFile.writeText("Hello, World!", Charsets.UTF_8)
+            Files.delete(dstFile)
+
+            val dstRelative = Paths.get(dstFile.fileName.toString())
+
+            srcFile.copyTo(dstRelative)
+
+            assertEquals(srcFile.readText(), dstFile.readText())
+        } finally {
+            Files.delete(dstFile)
+            Files.delete(srcFile)
+        }
+    }
+
+    private fun compareFiles(src: Path, dst: Path, message: String? = null) {
+        assertTrue(dst.exists())
+        assertEquals(src.isRegularFile(), dst.isRegularFile(), message)
+        if (dst.isRegularFile()) {
+            assertTrue(src.readBytes().contentEquals(dst.readBytes()), message)
+        }
+    }
+
+    @Test
+    fun testBufferedReader() {
+        val file = Files.createTempFile(null, null)
+        val lines = listOf("line1", "line2")
+        Files.write(file, lines)
+
+        assertEquals(file.bufferedReader().use { it.readLines() }, lines)
+        assertEquals(file.bufferedReader(StandardOpenOption.READ).use { it.readLines() }, lines)
```

Comment:
Could you check these overloads' usage in tests? 
I suppose they should be updated after we've removed all except one overload of each.

Reply:
I removed the lines testing the removed overloads and left tests for calls with the default argument and with all argument specified. Let me know if you want something else.